lib/Internal/HPack.php

```diff
@@ -75,43 +75,67 @@ class HPack {
         /* end! */ 30
     ];
 
-    private static $huffman_lookup;
-    private static $huffman_codes;
-    private static $huffman_lens;
+    private static $huffmanLookup;
+    private static $huffmanCodes;
+    private static $huffmanLengths;
 
+    private static $indexMap = [];
+
+    /** @var string[][] */
     private $headers = [];
+
+    /** @var int Max table size. */
     private $maxSize = 4096;
+
+    /** @var int Current table size. */
     private $size = 0;
 
     public static function init() {
-        self::$huffman_lookup = self::huffman_lookup_init();
-        self::$huffman_codes = self::huffman_codes_init();
-        self::$huffman_lens = self::huffman_lens_init();
+        self::$huffmanLookup = self::huffmanLookupInit();
+        self::$huffmanCodes = self::huffmanCodesInit();
+        self::$huffmanLengths = self::huffmanLengthsInit();
+
+        foreach (\array_column(self::TABLE, 0) as $index => $name) {
+            if (isset(self::$indexMap[$name])) {
+                continue;
+            }
+
+            self::$indexMap[$name] = $index + 1;
```

Comment:
Why wasn't this present before?

Reply:
Looks like I missed that one small optimization ;-)

Reply:
Technically the map should be to an array of potential header values, as some of the headers are in the table twice. As a server we really only stand to benefit from the predefined values for `:status`, which are tiny anyway, so there's not a lot to be gained, but something that could be improved upon further.